optuna/_hypervolume/hssp.py

```diff
@@ -44,10 +44,12 @@ def _solve_hssp_on_unique_loss_vals(
     subset_size: int,
     reference_point: np.ndarray,
 ) -> np.ndarray:
-    assert not np.any(reference_point - rank_i_loss_vals <= 0)
+    if not np.isfinite(reference_point).all():
+        return rank_i_indices[list(range(subset_size))]
```

Comment:
In this function, the length of `rank_i_indices` is guaranteed to be larger than or equal to `subset_size`, so the following should be fine.
```suggestion
        return rank_i_indices[:subset_size]
```

Reply:
Good catch! Thank you